### PR TITLE
Selection migration utility

### PR DESCRIPTION
**Pull request by skoszuta**

### CODE REVIEW

**Review by tomusdrw — APPROVED** [review]
Awesome!

File: packages/migrate-selection/index.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
+import type { ISelectionParams, ISynctexBlock, ISynctexBlockId, ISynctexData } from "@fluffylabs/types";
+import * as levenshtein from "fastest-levenshtein";
+
+const MULTI_LINE_BLOCK_PATTERN = /^\\begin{(.*?)}(.*?)^\\end{\1}/gms;
+const MIN_CONFIDENCE = 0.8;
+
+function getSiblingBlocks(synctex: ISynctexData, fileId: number, line: number): ISynctexBlock[] {
+  const siblingBlocks: ISynctexBlock[] = [];
+
+  for (const blocks of Object.values(synctex.pages)) {
```
[the remark this answers is not in the document]
Comment (skoszuta) [reply packages/migrate-selection/index.ts:10]: good point, will fix

File: src/components/CodeSyncProvider/hooks/useCodeStore.ts [hunk 2]
```diff
@@ -1,30 +1,53 @@
...
-  };
+    return {
+      async getTexAsString(path: string, version?: string): Promise<string> {
+        return fetchTex(path, version);
+      },
+      async getTexAsLines(path: string, version?: string): Promise<string[]> {
+        return (await fetchTex(path, version)).split("\n");
+      },
+      async getSynctex(version: string): Promise<ISynctexData> {
+        return JSON.parse(await fetchSynctex(version)) as ISynctexData;
```
[the remark this answers is not in the document]
Comment (skoszuta) [reply src/components/CodeSyncProvider/hooks/useCodeStore.ts:49]: good point as well

File: src/components/NoteManager/Note.tsx [hunk 3]
```diff
@@ -77,7 +78,16 @@ export function Note({ note, onEditNote, onDeleteNote, version }: NoteProps) {
...
+      {noteMigrated ? (
+        <NoteLink
+          note={note as IHighlightNote}
+          noteMigrated={noteMigrated as IHighlightNote}
+          version={version}
+          onEditNote={onEditNote}
+        />
+      ) : (
+        <>Loading...</>
+      )}
```
[the remark this answers is not in the document]
Comment (skoszuta) [reply src/components/NoteManager/Note.tsx]: done ✔️

File: src/components/CodeSyncProvider/hooks/useCodeStore.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (skoszuta) [reply src/components/CodeSyncProvider/hooks/useCodeStore.ts]: done ✔️

File: packages/migrate-selection/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (skoszuta) [reply packages/migrate-selection/index.ts]: done ✔️

**Pull request merged**